packages/belos/tpetra/test/LinearSolverFactory/SolverFactory.cpp

```diff
@@ -186,8 +187,14 @@ testSolver (Teuchos::FancyOStream& out,
 
   out << "Apply solver to \"solve\" AX=B for X.  Belos already has solver "
     "tests; the point is to check that solve() doesn't throw." << endl;
-  X->putScalar (STS::zero ());
-  solver->solve ();
+  try {
+    solver->solve ();
+  } catch (std::exception& e) {
+    out << "*** FAILED: Belos::SolverFactory::solve threw an exception: "
```

Comment:
`SolverManager::solve`?

Reply:
Added a try/catch loop around the solve because the test was not catching solver failures.  This was useful for debugging the test and solvers that were having issues with identity matrices.